impeller/renderer/backend/vulkan/allocator_vk.cc

```diff
@@ -15,6 +15,43 @@
 
 namespace impeller {
 
+// Maximum size to use VMA image suballocation. Any allocation greater than or
+// equal to this value will use a dedicated VkDeviceMemory.
+constexpr size_t kImageSizeThresholdForDedicatedMemoryAllocation =
+    4 * 1024 * 1024;
+
+static bool CreateBufferPool(VmaAllocator allocator, VmaPool* pool) {
```

Comment:
IMO, we should use fml::UniqueObject for pools, allocators, textures, and buffers from VMA. I took a closer look at the early returns and it seems to be fine but its easy to mess up and leak resources especially when it comes to error conditions. And we already have an RAII wrapper.

In a later patch is fine of course. Of file a followup and we can pick it up later so we don't forget.